### PR TITLE
sprite browser ease of use improvements

### PR DESCRIPTION
**Pull request by ericoporto**
same as #1344 

- Allows changing preview sizes for 1x, 2x, 3x, and 4x on right click;
- Slider gives 0.5 steps from 0.5x to 4x;
- Ctrl+Mouse Wheel also modifies the slider;
- Default size is 1x;
- Ctrl+0 resets to default size;
- Added a button for importing new sprite files;
- Image File Drop on Sprite Manager is supported, just click hold the desired image files on Windows Explorer and drag them in the Sprite Manager for import.

Droping files from Windows Explorer should not hang anymore. I can rebase and squash commits if needed.

### CODE REVIEW

**Comment by ericoporto** [comment]
I think dropped image files are getting corrupted, but I don't know why yet.

**Comment by ivan-mogilko** [comment]
> I think dropped image files are getting corrupted, but I don't know why yet.

Testing again, I have not noticed any corruption. What kind of files were causing it?

**Comment by ericoporto** [comment]
This is very curious, but I don't have the problem with the build from cirrus-ci, and building again today I don't have it locally too. 
Yesterday, my last sprite added was not being saved, so I was getting a blue cup when loading the project for the _last sprite I had added_.

**Review by persn — APPROVED** [review]
This looks good I think. I haven't tested it that it works, but the changes are simple enough that they should be valid. I left some minor feedback on variable naming and such

File: Editor/AGS.Editor/Panes/SpriteSelector.cs [hunk 1]
```diff
@@ -1223,6 +1262,16 @@ private void ShowSpriteContextMenu(Point menuPosition)
             menu.Items.Add(new ToolStripSeparator());
             menu.Items.Add(new ToolStripMenuItem("Find sprite by number...", null, onClick, MENU_ITEM_FIND_BY_NUMBER));
 
+            ToolStripMenuItem view_menu = new ToolStripMenuItem();
```
[the remark this answers is not in the document]
Comment (ericoporto) [reply Editor/AGS.Editor/Panes/SpriteSelector.cs]: Redid commit with style change. https://github.com/adventuregamestudio/ags/pull/1361/commits/92fe13961bcb03e9ec30f0b7f20151d9b03fecf2

File: Editor/AGS.Editor/Panes/SpriteSelector.cs [hunk 2]
```diff
@@ -214,7 +224,7 @@ private void DisplaySpritesForFolder(SpriteFolder folder)
             spriteList.Clear();
             _spriteImages.Images.Clear();
             _spriteImages.ColorDepth = ColorDepth.Depth16Bit;
-            _spriteImages.ImageSize = new Size(64, 64);
+            _spriteImages.ImageSize = new Size(32 * _spriteSizeMultiplier, 32 * _spriteSizeMultiplier);
```
[the remark this answers is not in the document]
Comment (ericoporto) [reply Editor/AGS.Editor/Panes/SpriteSelector.cs]: I now introduce SPRITE_BASE_SIZE here https://github.com/adventuregamestudio/ags/pull/1361/commits/92fe13961bcb03e9ec30f0b7f20151d9b03fecf2 and update it's value from 64 to 32 here https://github.com/adventuregamestudio/ags/pull/1361/commits/234e640d63c864503ba49b1cb5d5936f6f1eb516

File: Editor/AGS.Editor/Panes/SpriteSelector.cs [hunk 3]
```diff
@@ -225,7 +235,10 @@ private void DisplaySpritesForFolder(SpriteFolder folder)
             {
                 progress.SetProgressValue(index);
                 Sprite sprite = folder.Sprites[index];
-                Bitmap bmp = Utilities.GetBitmapForSpriteResizedKeepingAspectRatio(sprite, 64, 64, false, true, Color.Pink);
+
+                int new_size = Math.Min(Math.Max(Math.Max(sprite.Width, sprite.Height), 32), 32 * _spriteSizeMultiplier);
```
[the remark this answers is not in the document]
Comment (ericoporto) [reply Editor/AGS.Editor/Panes/SpriteSelector.cs]: I fixed the name style of this here https://github.com/adventuregamestudio/ags/pull/1361/commits/234e640d63c864503ba49b1cb5d5936f6f1eb516

File: Editor/AGS.Editor/Panes/SpriteSelector.Designer.cs [hunk 4]
```diff
@@ -28,84 +28,190 @@ protected override void Dispose(bool disposing)
...
 
         private System.Windows.Forms.SplitContainer splitWindow;
         private System.Windows.Forms.TreeView folderList;
+        private System.Windows.Forms.SplitContainer splitContainer1;
         private System.Windows.Forms.ListView spriteList;
+        private System.Windows.Forms.Panel panel1;
+        private System.Windows.Forms.TrackBar sliderPreviewSize;
+        private System.Windows.Forms.Button button_importNew;
+        private System.Windows.Forms.Label label1;
+        private System.ComponentModel.BackgroundWorker backgroundWorker1;
```
[the remark this answers is not in the document]
Comment (ericoporto) [reply Editor/AGS.Editor/Panes/SpriteSelector.Designer.cs]: changed the name to asyncFileDropWorker (https://github.com/adventuregamestudio/ags/pull/1361/commits/7a4e3d3a2094fb970401ac0247ac0f6714fe5956)

**Comment by ericoporto** [comment]
@persn thank you for the review! I tried to edit the original commits when possible to keep the history alright. 😃

**Comment by ivan-mogilko** [comment]
I think the one last proposal I have is to make key shortcuts go in a simple sequence:
Ctrl + 1 - x1 scale,
Ctrl + 2 - x2 scale - and so on,
because right now the sequence is separated into two keyboard sides: Ctrl + 0 sets to "default", Ctrl + 1 sets to x2 for some reason and so on.

**Comment by ericoporto** [comment]
Ctrl+0 corresponds to what today is the default size, this is the behavior that I am most familiar. For reference, in chrome browser, ctrl+0 resets zoom level, and ctrl+1-9 switches tabs.

It works like this, current reference size for icons (`SPRITE_BASE_SIZE`) is 64px. So current default zoom is 1*64px = 64px

This PR changes the `SPRITE_BASE_SIZE` to 32px. So current zoom is now 2*32px - this is what Ctrl+0 is supposed to reset to, so this is why it set multiplier to 2. 

So should I make ctrl+0 and ctrl+1 do the same thing?

**Comment by ivan-mogilko** [comment]
Well the question is whether there should be shortcuts for individual zoom levels, or only for default level.
If there are shortcuts for individual zoom levels, then it's only natural in my opinion that the key number corresponds to a zoom level, and the shortcuts go in one sequence.

**Comment by ericoporto** [comment]
erh, I opted to remove the ctrl+1, ctrl+2, ctrl+3 shortcut for now - it's easier to add back in the future if someone mentions it...

**Pull request merged**